app/addons/documents/resources.js

```diff
@@ -434,5 +434,43 @@ function (app, FauxtonAPI, Documents, PagingCollection) {
     }, menuLinks);
   };
 
+  Documents.RevTreeDataModel = Backbone.Model.extend({
+    initialize: function (url) {
+      this.url = url;
+      console.log("inside initialize");
+    },
+    url: function () {
+      return this.url;
+    },
+
+    sync: function (method, model, options) {
+      console.log("inside Sync-----" + this.url);
+      var params = {
+        error: options.error,
+        success: options.success,
+        url: this.url,
+        type: 'GET',
+        dataType: 'text',
+        async: false
+      };
+
+      return $.ajax(params);
+    },
+    parse: function (response) {
+
+      var parsedResult = [];
+      var splitResponse = response.split(/(\n|\r\n|\r)/);
+
+      for (var i = 0; i < splitResponse.length; i++) {
+        if (String(splitResponse[i]).charAt(0) == "{") {
+          parsedResult.push(JSON.parse(splitResponse[i]));
+        }
+      }
+      console.log(parsedResult);
+
+      return {content: parsedResult};
```

Comment:
this code needs some tests, example test: https://github.com/apache/couchdb-fauxton/blob/master/app/addons/documents/tests/resourcesSpec.js#L45-72